core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -1179,4 +1238,61 @@ public void setNext(int next) {
             cachedNext = next;
         }
     }
+
+    public static class ShareFetchPartitionData {
+        private final FetchParams fetchParams;
+        private final String groupId;
+        private final String memberId;
+        private final Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData;
+        private final CompletableFuture<Map<TopicIdPartition, ShareFetchResponseData.PartitionData>> future;
+
+        public ShareFetchPartitionData(FetchParams fetchParams, String groupId, String memberId,
+                                       Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData,
+                                       CompletableFuture<Map<TopicIdPartition, ShareFetchResponseData.PartitionData>> future) {
+            this.fetchParams = fetchParams;
+            this.groupId = groupId;
+            this.memberId = memberId;
+            this.topicPartitionData = topicPartitionData;
+            this.future = future;
+        }
+
+        public FetchParams fetchParams() {
+            return fetchParams;
+        }
+
+        public String groupId() {
+            return groupId;
+        }
+
+        public String memberId() {
+            return memberId;
+        }
+
+        public Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData() {
+            return topicPartitionData;
+        }
+
+        public CompletableFuture<Map<TopicIdPartition, ShareFetchResponseData.PartitionData>> future() {
+            return future;
+        }
```

Comment:
May be have them if required in `tests` else we can access the variables directly (private variables in same parent class can be accessed). Not a fan of `getters` and we might learn more about the semantics with AK reviewers as they can point what we follow in AK code base.